pkg/healthcheck/healthcheck_output.go

```diff
@@ -396,6 +385,26 @@ func restartSpinner(spin *spinner.Spinner, result *CheckResult) {
 	}
 }
 
+func printHeader(wout io.Writer, headerPrinted bool, hc Runner) bool {
```

Comment:
This bit is confusing so I think it would benefit from a comment explaining how this works.  Namely that when running in short mode, we defer writing the header until the first time we print a result.